server/routes/chat_routes.ts

```diff
@@ -350,11 +374,18 @@ export function registerChatRoutes(router: IRouter, routeOptions: RoutesOptions)
        * Retrieve latest interactions from memory
        */
       try {
-        const conversation = await storageService.getSession(sessionId);
+        const interaction = await storageService.getInteraction(
+          sessionId,
+          outputs?.interactionId || ''
+        );
+        const finalInteractions = [interaction].filter((item) => item);
+        const messages =
+          (await storageService.getMessagesFromInteractions(finalInteractions)) || [];
 
         return response.ok({
           body: {
-            ...conversation,
+            interactions: [interaction],
```

Comment:
Shall we change to `interactions: finalInteractions` here?

Reply:
Sure, done